linkis-public-enhancements/linkis-bml/linkis-bml-client/src/main/java/com/webank/wedatasphere/linkis/bml/client/BmlClientFactory.java

```diff
@@ -16,22 +16,28 @@
 package com.webank.wedatasphere.linkis.bml.client;
 
 import com.webank.wedatasphere.linkis.bml.client.impl.HttpBmlClient;
+import com.webank.wedatasphere.linkis.bml.http.HttpConf;
 
 import java.util.Map;
 
 public class BmlClientFactory {
     public static BmlClient createBmlClient(){
-        return createBmlClient(null, null);
+        return createBmlClient(null, null, null);
     }
 
 
     public static BmlClient createBmlClient(String user){
-        return createBmlClient(user, null);
+        return createBmlClient(user, null, null);
     }
 
-
     public static BmlClient createBmlClient(String user, Map<String, Object> properties){
-        AbstractBmlClient bmlClient = new HttpBmlClient();
+        return createBmlClient(user, properties, null);
+    }
+    public static BmlClient createBmlClient(String user, Map<String, Object> properties, String serverUrl){
+        if (serverUrl == null || "".equals(serverUrl.trim())) {
+            serverUrl = HttpConf.gatewayInstance();
+        }
+        AbstractBmlClient bmlClient = new HttpBmlClient(serverUrl);
         bmlClient.setUser(user);
         bmlClient.setProperties(properties);
```

Comment:
The passed-in properties are currently not valid. Can you optimize this place by the way so that when HttpBmlClient creates a clientConfig, some parameters can be obtained from properties field?

Reply:
Ok i will optimize it.